python/res/enkf/config/enkf_config_node.py

```diff
@@ -144,6 +229,147 @@ def create_gen_data(cls, key, file_fmt, report_steps=(0,)):
+                                             init_file_fmt)
+        if config_node is None:
+            raise ValueError("Failed to create GEN KW node for:%s" % key)
+
+        return config_node
+
+    # SURFACE FULL creation
+    @classmethod
+    def create_surface(cls, key, init_file_fmt, output_file, base_surface_file, min_std_file, forward_init):
+
+        base_surface_file = cls.get_real_path(base_surface_file)
+        config_node = cls._alloc_surface_full(key,
+                                              forward_init,
+                                              output_file,
+                                              base_surface_file,
+                                              min_std_file,
+                                              init_file_fmt)
+        if config_node is None:
+            raise ValueError("Failed to create SURFACE node for:%s" % key)
+
+        return config_node
+
+    # FIELD FULL creation
+    @classmethod
+    def create_field(cls, key, var_type_string, grid, field_trans_table, ecl_file, enkf_infile, forward_init,
+                     init_transform, output_transform, input_transform, min_std_file, min_key, max_key, init_file_fmt):
+
+        truncation = EnkfTruncationType.TRUNCATE_NONE
+        value_min = -1
+        value_max = -1
+        if min_key is not None:
+            value_min = min_key
+            truncation = truncation | EnkfTruncationType.TRUNCATE_MIN
+        if max_key is not None:
+            value_max = max_key
+            truncation = truncation | EnkfTruncationType.TRUNCATE_MAX
+        
+        config_node = cls._alloc_field_node(key, grid, field_trans_table, forward_init)
+        if config_node is not None and var_type_string == "PARAMETER":
+            config_node._update_parameter_field(ecl_file, 
+                                                init_file_fmt,
+                                                min_std_file,
+                                                truncation,
+                                                value_min,
+                                                value_max,
+                                                init_transform,
+                                                output_transform)
+
+        elif config_node is not None and var_type_string == "GENERAL":
+            config_node._update_general_field(ecl_file,
+                                              enkf_infile,
+                                              init_file_fmt,
+                                              min_std_file,
+                                              truncation,
+                                              value_min,
+                                              value_max,
+                                              init_transform,
+                                              input_transform,
+                                              output_transform)
+
+        if config_node is None:
```

Comment:
This test could be moved up prior to ` if config_node is not None and var_type_string == "PARAMETER":` and thus remove the need for testing config_node at a later stage.

Reply:
Agree